dags/replicator_table_check.py

```diff
@@ -0,0 +1,161 @@
+- `not_up_to_date`: tables which are being erroneously copied from `move_staging` without being up to date.
+- `outdated_remove`: tables in `move_staging` which are outdated.
+"""
+
+#!/data/airflow/airflow_venv/bin/python3
+# -*- coding: utf-8 -*-
+# noqa: D415
+import os
+import sys
+import pendulum
+# pylint: disable=import-error
+from airflow.decorators import dag, task
+from airflow.models import Variable
+from airflow.exceptions import AirflowFailException
+from airflow.providers.postgres.hooks.postgres import PostgresHook
+
+# import custom operators and helper functions
+repo_path = os.path.abspath(os.path.dirname(os.path.dirname(os.path.realpath(__file__))))
+sys.path.insert(0, repo_path)
+# pylint: disable=wrong-import-position
+from dags.dag_functions import task_fail_slack_alert
+from dags.common_tasks import wait_for_external_trigger
+# pylint: enable=import-error
+
+DAG_NAME = 'replicator_table_check'
+DAG_OWNERS = Variable.get("dag_owners", deserialize_json=True).get(DAG_NAME, ["Unknown"])
+
+default_args = {
+    "owner": ",".join(DAG_OWNERS),
+    "depends_on_past": False,
+    "start_date": pendulum.datetime(2024, 4, 12, tz="America/Toronto"),
+    "email_on_failure": False,
+    "retries": 0,
+    "on_failure_callback": task_fail_slack_alert,
+}
+
+@dag(
+    dag_id=DAG_NAME,
+    default_args=default_args,
+    catchup=False,
+    max_active_runs=1,
+    schedule=None, #triggered externally
+    doc_md=__doc__,
+    tags=["replicator", "data_checks"]
+)
+def replicator_DAG():
+
+    @task()
+    def tables_to_copy():
+        """This task finds all the replicators from `replicators` airflow variable,
+        and then finds all the tables listed for replication by looking at the Airflow variable
+        listed in the `tables` key item in the replicator dictionaries."""
+
+        #a list of the replicators
+        replicators = Variable.get('replicators', deserialize_json=True)
+    
+        #extract source tables from airflow variables
+        tables_to_copy = []
+        for _, dag_items in replicators.items():
+            tables = Variable.get(dag_items['tables'], deserialize_json=True)
+            src_tables = [tbl[0] for tbl in tables]
+            tables_to_copy = tables_to_copy + src_tables
+
+        #get only source table names
+        return tables_to_copy
+
+    @task()
+    def updated_tables(ds):
+        """This task finds tables in `move_staging` with comments
+        indicating they are up to date ("last updated on {ds}")."""
+
+        updated_tables_sql = """
+        SELECT pgn.nspname::text || '.' || pgc.relname::text, pgd.description
+        FROM pg_description AS pgd
+        JOIN pg_class AS pgc ON pgd.objoid = pgc.oid
+        JOIN pg_namespace pgn ON pgc.relnamespace = pgn.oid
+        WHERE
+            pgn.nspname = 'move_staging'
+            AND pgc.relkind = 'r'
+            AND pgd.description ILIKE %s"""
+
+        con = PostgresHook("collisions_bot").get_conn()
```

Comment:
is it all collision related tables? Should we use replicator_bot instead?

Reply:
It's both collisions and counts tables within `move_staging`, but from my testing, using this method works even if you don't have correct select permissions. 

Tested as `vds_bot`: 

Returns 62 rows:
```sql
SELECT pgn.nspname::text || '.' || pgc.relname::text, pgd.description
FROM pg_description AS pgd
JOIN pg_class AS pgc ON pgd.objoid = pgc.oid
JOIN pg_namespace pgn ON pgc.relnamespace = pgn.oid
WHERE
    pgn.nspname = 'move_staging'
    AND pgc.relkind = 'r'
```

Original method using `obj_description()` returns 0 rows:
```sql
WITH move_staging_comments AS (
    SELECT table_schema::text || '.' || table_name::text AS tbl_name
    FROM information_schema.tables
    WHERE table_schema = 'move_staging'
)

SELECT tbl_name, obj_description(tbl_name::regclass) 
FROM move_staging_comments
```

So it doesn't really matter which connection we use here.

Reply:
not really on the permission issue but more on using the collision bot for only collision related tasks, since they all have limited connection (granted it has 20 :meow_dio:) and we already have a replicator bot